tests/test_dataset_cards.py

```diff
@@ -0,0 +1,100 @@
+# coding=utf-8
+# Copyright 2021 HuggingFace Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from pathlib import Path
+from subprocess import check_output
+from typing import List
+
+import pytest
+
+from datasets.utils.logging import get_logger
+from datasets.utils.metadata import DatasetMetadata
+from datasets.utils.readme import ReadMe
+
+from .utils import slow
+
+
+repo_path = Path.cwd()
+
+logger = get_logger(__name__)
+
+
+def get_changed_datasets(repo_path: Path) -> List[Path]:
+    diff_output = check_output(["git", "diff", "--name-only", "HEAD..origin/master"], cwd=repo_path)
+    changed_files = [Path(repo_path, f) for f in diff_output.decode().splitlines()]
+
+    changed_datasets = set(
+        f.parent.parts[-1]
+        for f in changed_files
+        if f.exists() and f.parent.parent.name == "datasets" and f.parent.parent.parent.name == "datasets"
+    )
+
+    return sorted(changed_datasets)
+
+
+def get_all_datasets(repo_path: Path) -> List[Path]:
+    return [path.parts[-1] for path in (repo_path / "datasets").iterdir()]
+
+
+@slow
+@pytest.mark.parametrize("dataset_name", get_all_datasets(repo_path))
+def test_dataset_tags(dataset_name):
+    card_path = repo_path / "datasets" / dataset_name / "README.md"
+    assert os.path.exists(card_path)
+    DatasetMetadata.from_readme(card_path)
+
+
+@pytest.mark.parametrize("dataset_name", get_changed_datasets(repo_path))
+def test_changed_dataset_tags(dataset_name):
+    card_path = repo_path / "datasets" / dataset_name / "README.md"
+    assert os.path.exists(card_path)
+    DatasetMetadata.from_readme(card_path)
+
+
+@slow
+@pytest.mark.parametrize("dataset_name", get_all_datasets(repo_path))
+def test_readme_content(dataset_name):
+    card_path = repo_path / "datasets" / dataset_name / "README.md"
+    assert os.path.exists(card_path)
+    ReadMe.from_readme(card_path)
+
+
+@pytest.mark.parametrize("dataset_name", get_changed_datasets(repo_path))
+def test_changed_readme_content(dataset_name):
+    card_path = repo_path / "datasets" / dataset_name / "README.md"
+    assert os.path.exists(card_path)
+    ReadMe.from_readme(card_path)
+
+
+@slow
+@pytest.mark.parametrize("dataset_name", get_all_datasets(repo_path))
+def test_dataset_card(dataset_name):
```

Comment:
We can remove this test since it does the same thing as `test_dataset_tags` and `test_readme_content `

Reply:
Didn't we want to combine the test? I assume the user would want to run a single test locally.

Reply:
You can definitely combine them. As you say it will be more convenient for users to run them locally.
In this case maybe let's only keep `test_dataset_card` and remove `test_dataset_tags` and `test_readme_content` ?

The same goes changed datasets: we can have `test_changed_dataset_card` and remove `test_changed_dataset_tags ` and `test_changed_readme_content` in order to not have duplicates.